src/js/stores/MesosStateStore.js

```diff
@@ -99,14 +99,15 @@ class MesosStateStore extends GetSetBaseStore {
     // refresh limits to the UI. They are:
     //
     // MOST once every (Config.getRefreshRate() * 0.5) ms. due to debounceTime.
-    // LEAST once every tick of Config.getRefreshRate() ms in Observable.interval
+    // LEAST once every tick of Config.getRefreshRate() ms in
+    // Observable.interval
     //
     // TODO: https://jira.mesosphere.com/browse/DCOS-18277
     this.stream = waitStream
       .concat(eventTriggerStream)
       .debounceTime(Config.getRefreshRate() * 0.5)
       .retryWhen(linearBackoff(MAX_RETRIES, RETRY_DELAY))
-      .subscribe(this.onStreamData, this.onStreamError);
+      .subscribe(Promise.resolve().then(this.onStreamData), this.onStreamError);
```

Comment:
any reason you chose `Promise` over `setTimeout`? Might be personal taste, but I'd argue `setTimeout` is adds a bit less complexity here. specially because you dont `catch`.

Reply:
Would it make more sense wrapping the whole `subscribe` method in `addChangeListener`. Those integration tests bug me, not reproducible locally though.

Reply:
found this on why the promise approach:
https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context

Reply:
@juliangieseke I went with `Promise.resolve().then()` as it schedules microtasks which are executed right after the script is done but before the next tasks are started. Using `setTimeout` would schedule a new task which is not necessary in this case. 

I recommend the following post for details on tasks and microtasks: https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/ 

@nLight What do you mean with "wrapping the whole subscribe method in addChangeListener"?

Reply:
Wouldn't this do the same trick and stop exception propagation?

```js
  addChangeListener(eventName, callback) {
    this.on(eventName, callback);

    if (this.shouldSubscribe()) {
        Promise.resolve().then(() => this.subscribe());
    }
```

Reply:
Regarding micro/macro tasks and observer scheduler, we settled that in this case it does not really matter for us, but I would like to share a little bit of what I learned about schedulers.

Contrary to rxjs 4, rxjs 5 default scheduler depends on the operator.
https://github.com/ReactiveX/rxjs/blob/master/doc/scheduler.md

The default scheduler for `from` rxjs 5 is synchronous `Scheduler.queue`:
https://github.com/Reactive-Extensions/RxJS/blob/master/src/modular/observable/from.js
https://jsbin.com/keriluy/edit?js,console

For `Interval` on the other hand, it is `Scheduler.async`:
https://github.com/Reactive-Extensions/RxJS/blob/master/src/modular/observable/interval.js

Reply:
The `.subscribe(Promise.resolve().then....)` breaks the nodes page (my guess because that page stops listening). `Promise.resolve().then(this.subscribe)` (by @nLight), contrary to what I said, fix the issue without causing the nodes page to break. 

I am switching to that fix.